eradiate/scenes/atmosphere/_molecules.py

```diff
@@ -0,0 +1,438 @@
+        return to_quantity(self.thermoprops.z_level).max()
+
+    @property
+    def thermoprops(self) -> xr.Dataset:
+        return self._thermoprops
+
+    def eval_width(self, ctx: Optional[KernelDictContext] = None) -> pint.Quantity:
+        if self.width is AUTO:
+            spectral_ctx = ctx.spectral_ctx if ctx is not None else None
+            min_sigma_s = self.radprops_profile.eval_sigma_s(spectral_ctx).min()
+
+            if min_sigma_s <= 0.0:
+                raise ValueError(
+                    "cannot compute width automatically when scattering "
+                    "coefficient reaches zero"
+                )
+
+            return min(10.0 / min_sigma_s, ureg.Quantity(1e3, "km"))
+        else:
+            return self.width
+
+    # --------------------------------------------------------------------------
+    #                             Radiative properties
+    # --------------------------------------------------------------------------
+
+    @property
+    def radprops_profile(self) -> RadProfile:
+        if self.thermoprops.title == "U.S. Standard Atmosphere 1976":
+            if self.absorption_data_sets is not None:
+                absorption_data_set = self.absorption_data_sets["us76_u86_4"]
+            else:
+                absorption_data_set = None
+            return US76ApproxRadProfile(
+                thermoprops=self.thermoprops,
+                has_scattering=self.has_scattering,
+                has_absorption=self.has_absorption,
+                absorption_data_set=absorption_data_set,
+            )
+        elif "AFGL (1986)" in self.thermoprops.title:
+            return AFGL1986RadProfile(
+                thermoprops=self.thermoprops,
+                has_scattering=self.has_scattering,
+                has_absorption=self.has_absorption,
+                absorption_data_sets=self.absorption_data_sets,
+            )
+        else:
+            raise NotImplementedError("Unsupported thermophysical properties data set.")
+
+    def eval_radprops(self, spectral_ctx: SpectralContext) -> xr.Dataset:
+        """
+        Evaluates the molecular atmosphere's radiative properties.
+
+        Parameter ``spectral_ctx`` (:class:`SpectralContext`):
+            Spectral context.
+
+        Returns → :class:`~xarray.Dataset`:
+            Radiative properties dataset.
+        """
+        return self.radprops_profile.to_dataset(spectral_ctx=spectral_ctx)
+
+    # --------------------------------------------------------------------------
+    #                             Kernel dictionary
+    # --------------------------------------------------------------------------
+
+    @property
+    def albedo_file(self) -> pathlib.Path:
+        return self.cache_dir / self.albedo_filename
+
+    @property
+    def sigma_t_file(self) -> pathlib.Path:
+        return self.cache_dir / self.sigma_t_filename
+
+    def kernel_phase(self, ctx: Optional[KernelDictContext]) -> MutableMapping:
```

Comment:
You can hint a parameter as `Optional` **iff** it has `None` as its default (other occurences below).